### DISCUSSION TITLE
How to include DevTools in production

### DISCUSSION THREAD

**Discussion by ktw1016** [TanStack/query#1981] [Q&A]
I need to include DevTools in production, I tried following deprecated docs [here](https://github.com/tannerlinsley/react-query-devtools#usage) by importing `dist/react-query-devtools.production.min.js` file directly but get errors.

I couldn't find in the doc or in source code a way to include DevTools in production. Are there ways to do this? If not, could you guys include that as a feature?

**Comment by TkDodo** [comment]
hm, I can still see `react-query/dist/react-query-devtools.production.min.js` what error are you getting if you try to import it like that?

**Reply by ktw1016** [reply]
Hey @TkDodo thanks for the quick response! I should've included that right away. When I `import { ReactQueryDevtools } from "./node_modules/react-query/dist/react-query-devtools.production.min.js";`, I get:
<img width="613" alt="Screen Shot 2021-03-16 at 3 16 58 PM" src="https://user-images.githubusercontent.com/29746626/111367132-cb46c400-866a-11eb-9a74-9e4254efc694.png">

**Reply by TkDodo** [reply]
hmmm, there is an open issue regarding the devtools getting their own version of react-query and the queryClient provider thus not picking up the right context, see: https://github.com/tannerlinsley/react-query/issues/1936

**Reply by ktw1016** [reply]
I re-installed `react-query@3.12.0` and `import { ReactQueryDevtools } from "./node_modules/react-query/dist/react-query-devtools.production.min.js"` works! I'll use this as temporary workaround for now. Thanks!
